File: Atcoder_#343/javascript/b_adjacency_matrix.js
function main(input) {
  const lines = input.trim().split('\n');
  const n = parseInt(lines[0], 10);
  const adjacencyMatrix = lines.slice(1).map(line => line.split(' ').map(Number));

  for (let i = 0; i < n; i++) {
    const connectedVertices = [];

    for (let j = 0; j < n; j++) {
      if (adjacencyMatrix[i][j] === 1) {
        connectedVertices.push(j + 1); 
      }
    }

    console.log(connectedVertices.sort((a, b) => a - b).join(' '));
  }
}

main(require('fs').readFileSync('/dev/stdin', 'utf8'));

File: Atcoder_#343/ruby/b_adjacency_matrix.rb
def main
  n = gets.to_i
  adjacency_matrix = []

  n.times do
    adjacency_matrix << gets.split.map(&:to_i)
  end

  adjacency_matrix.each_with_index do |row, i|
    connected_vertices = []
    
    row.each_with_index do |value, j|
      if value == 1
        connected_vertices << (j + 1)
      end
    end

    puts connected_vertices.sort.join(" ")
  end
end

main
